tests/lib/prepare.sh

```diff
@@ -621,10 +621,16 @@ prepare_ubuntu_core() {
     fi
 
     echo "Ensure the core snap is cached"
+    # Cache snaps
     if is_core18_system; then
         if ! snap list core; then
             cache_snaps core
         fi
+        if snap list core; then
```

Comment:
This `if` and the one above cannot be true at the same time. Maybe refactor this to:
```
if is_core18_system; then
    if snap list core;
        echo "core snap on core18 should not be installed yet"
        snap list
        exit 1
    fi
    cache_snaps core
fi
```

Reply:
done

Reply:
Thanks for the review